### PR TITLE
Clear WPFS DB based on duration (hr, min, sec)

### PR DESCRIPTION
**Pull request by seswarrajan**
Signed-off-by: Eswar Rajan Subramanian <eswar@accuknox.com>

### CODE REVIEW

**Comment by seswarrajan** [comment]
To clear the WPFS DB based on duration, the script to be used in following format. 

```
./observe_clear_sys_data.sh --req dbclear --duration xxh
./observe_clear_sys_data.sh --req dbclear --duration xxm
./observe_clear_sys_data.sh --req dbclear --duration xxs

or 

./observe_clear_sys_data.sh --req dbclear --duration xxhxxmxxs
```

File: src/systempolicy/observability.go [hunk 1]
```diff
@@ -103,7 +106,44 @@ func GetSystemObsData(wpfs types.WorkloadProcessFileSet) (opb.Response, error) {
 	return opbSysObsResponse, nil
 }
 
-func ClearSysDb(wpfs types.WorkloadProcessFileSet, duration int64) error {
-	err := libs.ClearWPFSDb(CfgDB, wpfs, duration)
+func convertDurationStrToInt(durationStr string) int {
```
Comment (nyrahul) [inline src/systempolicy/observability.go]: we should not write the API ourself.

Please check https://golangrepo.com/repo/xhit-go-str2duration-go-date-time

**Review by nyrahul — CHANGES_REQUESTED** [review]

File: src/systempolicy/observability.go [hunk 2]
```diff
@@ -103,7 +106,11 @@ func GetSystemObsData(wpfs types.WorkloadProcessFileSet) (opb.Response, error) {
 	return opbSysObsResponse, nil
 }
 
-func ClearSysDb(wpfs types.WorkloadProcessFileSet, duration int64) error {
-	err := libs.ClearWPFSDb(CfgDB, wpfs, duration)
+func ClearSysDb(wpfs types.WorkloadProcessFileSet, durationStr string) error {
+	if durationStr == "0" {
+		return errors.New("not a valid duration")
+	}
+	duration, _ := time.ParseDuration(durationStr)
```
Comment (nyrahul) [inline src/systempolicy/observability.go]: Please handle this error.
```suggestion
	duration, err := time.ParseDuration(durationStr)
	if err != nil {
	     return err
	}
```

**Review by nyrahul — CHANGES_REQUESTED** [review]

File: src/systempolicy/observability.go [hunk 3]
```diff
@@ -103,7 +106,14 @@ func GetSystemObsData(wpfs types.WorkloadProcessFileSet) (opb.Response, error) {
 	return opbSysObsResponse, nil
 }
 
-func ClearSysDb(wpfs types.WorkloadProcessFileSet, duration int64) error {
-	err := libs.ClearWPFSDb(CfgDB, wpfs, duration)
+func ClearSysDb(wpfs types.WorkloadProcessFileSet, durationStr string) error {
+	if durationStr == "0" {
+		return errors.New("not a valid duration")
+	}
```
Comment (nyrahul) [inline src/systempolicy/observability.go:112]: What is the point of this check? 
If the format is invalid then time.ParseDuration() will return an err.
```suggestion
```

**Review by nyrahul — CHANGES_REQUESTED** [review]

File: src/systempolicy/observability.go (hunk 3 above)
Comment (seswarrajan) [reply src/systempolicy/observability.go:112]: Yes Rahul. But when we input 0, this is considered as a valid input and the output is given as 0 seconds.
That is why I explicitly included this check.

**Pull request merged**